lib/phonebook/providers/roles_tags_provider.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:myecl/auth/providers/openid_provider.dart';
+import 'package:myecl/phonebook/class/association.dart';
+import 'package:myecl/phonebook/class/complete_member.dart';
+import 'package:myecl/phonebook/repositories/role_tags_repository.dart';
+import 'package:myecl/tools/providers/map_provider.dart';
+import 'package:myecl/tools/token_expire_wrapper.dart';
+
+class RolesTagsNotifier extends MapNotifier<String, bool> {
+  final RolesTagsRepository rolesTagsRepository = RolesTagsRepository();
+  RolesTagsNotifier({required String token}) {
+    rolesTagsRepository.setToken(token);
+  }
+
+
+
+  // void setRole(RolesTags i) {
+  //   state = AsyncValue.data(Tuple2(i, state.value!.item2));
+  // }
+
+  Future<void> loadRolesTags() async {
+    loadTList([]);
+    final result = await rolesTagsRepository.getRolesTags();
+    for (int i = 0; i < result.tags.length; i++) {
+        setTData(result.tags[i], const AsyncData([false]));
+    }
+  }
+
+  void resetChecked() {
+    state.maybeWhen(data: (d) {
```

Comment:
If you only want to manipulate data without return anything, do it with whenData